src/main/java/teammates/ui/controller/InstructorCoursesPageData.java

```diff
@@ -7,29 +8,172 @@
 import teammates.common.datatransfer.InstructorAttributes;
 import teammates.common.datatransfer.CourseAttributes;
 import teammates.common.datatransfer.CourseDetailsBundle;
+import teammates.common.util.Const;
+import teammates.common.util.Sanitizer;
+import teammates.ui.template.ActiveCoursesTable;
+import teammates.ui.template.ActiveCoursesTableRow;
+import teammates.ui.template.ArchivedCoursesTable;
+import teammates.ui.template.ArchivedCoursesTableRow;
+import teammates.ui.template.ElementTag;
 
 /**
  * This is the PageData object for the 'Courses' page 
  */
 public class InstructorCoursesPageData extends PageData {
+    private ArchivedCoursesTable archivedCourses;
+    private ActiveCoursesTable activeCourses;
+    private String courseIdToShow;
+    private String courseNameToShow;
+    private HashMap<String, InstructorAttributes> instructorsForCourses;
     
     public InstructorCoursesPageData(AccountAttributes account) {
         super(account);
     }
     
-    /** Used when adding a course. Null if not adding a course. */
-    public CourseAttributes newCourse;
+    public void init(List<CourseDetailsBundle> activeCoursesParam, List<CourseDetailsBundle> archivedCoursesParam,
+                     HashMap<String, InstructorAttributes> instructorsForCoursesParam){
+        init(activeCoursesParam, archivedCoursesParam, instructorsForCoursesParam, "", ""); 
+    }
     
-    public HashMap<String, InstructorAttributes> instructors;
+    public void init(List<CourseDetailsBundle> activeCoursesParam, List<CourseDetailsBundle> archivedCoursesParam, 
+                     HashMap<String, InstructorAttributes> instructorsForCoursesParam, String courseIdToShowParam, String courseNameToShowParam){
+        instructorsForCourses = instructorsForCoursesParam;
+        activeCourses = convertToActiveCoursesTable(activeCoursesParam);
+        archivedCourses = convertToArchivedCoursesTable(archivedCoursesParam);
+        courseIdToShow = courseIdToShowParam;
+        courseNameToShow = courseNameToShowParam;
```

Comment:
I personally recommend using
````
this.instructorsForCourses = instructorsForCourses
````
instead of
````
instructorsForCourses = instructorForCoursesParam
````
May be just a preference, though.